### PR TITLE
Facet ux refresh

### LINKED ISSUE #1476: Refresh flex facet UX
https://docs.google.com/document/d/1rE4Ir9UnRXoLnRxxxqF63KmBL8xkl_W1KOwqUwdtM6Y/edit

### PR DESCRIPTION
**Pull request by etanshaul**
fixes #1476 

Unit test updates pending.

Can review code and updated UX (I'll post screenshots here).

### CODE REVIEW

**Comment by etanshaul** [comment]
New UX:

Initial state, no app.yaml or Dockerfile:
image

app.yaml generate dialog (notice if the path is missing in above screenshot, it defaults to the canonical dir):
image

after generating the app.yaml (it generates runtime: java):
image

after changing the runtime to custom:
image

Dockerfile generate dialog:
image

After generating a Dockerfile:
image

**Comment by etanshaul** [comment]
One important difference with these changes is that validation errors no longer prevent the user from saving the dialog.

**Comment by etanshaul** [comment]
@patflynn tests updated now too. Ready for review.

File: google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties [hunk 1]
```diff
@@ -74,9 +74,19 @@ appengine.flex.configtype.custom.label=Custom
 appengine.flex.user.specified.artifact.label=Select a deployment archive:
 appengine.flex.dockerfile.location.label=Dockerfile path\:
 appengine.flex.version.placeholder.text=auto generated
-appengine.flex.config.destination.folder.label=Choose destination folder\:
+appengine.flex.facet.config.title=App Engine flexible configuration
+appengine.flex.facet.config.appyaml.error.before.text=The specified app.yaml is invalid or does not exist. Browse to an app.yaml file, or 
+appengine.flex.facet.config.dockerfile.directory.error.before.text=The directory does not contain a Dockerfile. Browse to a directory with a Dockerfile, or 
+appengine.flex.facet.config.runtime.explanation.label=if you want to provide a Dockerfile, change the runtime to 'custom' in app.yaml
+appengine.flex.facet.config.appyaml.generate.link.text=generate a new app.yaml ...
```
Comment (patflynn) [inline google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: i have a doubt whether the name of the file should be a resource bundle or provided by a plugin level constant

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.form [hunk 2]
```diff
@@ -22,83 +22,158 @@
...
           </component>
-          <component id="34f49" class="javax.swing.JButton" binding="genAppYamlButton">
+          <component id="81422" class="javax.swing.JLabel">
             <constraints>
-              <grid row="1" column="0" row-span="1" col-span="1" vsize-policy="0" hsize-policy="1" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
+              <grid row="0" column="0" row-span="1" col-span="1" vsize-policy="0" hsize-policy="0" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
             </constraints>
             <properties>
-              <text value="Generate..."/>
+              <text value="app.yaml:"/>
```
Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.form]: could be nice to set this from the same global constant.

File: google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties (hunk 1 above)
Comment (etanshaul) [inline google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: the thing is though that the user can technically name it whatever they want. "app.yaml" here in the message isn't really the name of the file but just a convenient way to reference it. does your comment still apply given this?

Comment (etanshaul) [inline google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: actually, nm, as its currently implemented the user is forced to use "app.yaml"

Comment (etanshaul) [inline google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: @patflynn there are a lot of cases all over the source where "app.yaml" is used. Rather then polluting this PR I suggest we do this separately. I can file an issue.

Comment (etanshaul) [inline google-cloud-tools-plugin/resources/messages/CloudToolsBundle.properties]: https://github.com/GoogleCloudPlatform/google-cloud-intellij/issues/1483

**Pull request merged**